apps/web/src/components/Solana/ConnectWalletModal.tsx

```diff
@@ -0,0 +1,169 @@
+        notInstalled.push(wallet);
+      }
+    }
+
+    return [installed, notInstalled];
+  }, [wallets]);
+
+  const hideModal = useCallback(() => {
+    setFadeIn(false);
+    setTimeout(() => setVisible(false), 150);
+  }, [setVisible]);
+
+  const handleClose = useCallback(
+    (event: MouseEvent) => {
+      event.preventDefault();
+      hideModal();
+    },
+    [hideModal],
+  );
+
+  const handleWalletClick = useCallback(
+    (_event: MouseEvent, walletName: WalletName) => {
+      select(walletName);
+      // handleClose(event);
+    },
+    [select],
+  );
+
+  useEffect(() => {
+    if (selectedWallet && connected) {
+      hideModal();
+    }
+  }, [selectedWallet, connected, hideModal]);
+
+  useLayoutEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === "Escape") {
+        hideModal();
+      }
+    };
+
+    // Get original overflow
+    const { overflow } = window.getComputedStyle(document.body);
+    // Hack to enable fade in animation after mount
+    setTimeout(() => setFadeIn(true), 0);
+    // Prevent scrolling on mount
+    document.body.style.overflow = "hidden";
+    // Listen for keydown events
+    window.addEventListener("keydown", handleKeyDown, false);
+
+    return () => {
+      // Re-enable scrolling when component unmounts
+      document.body.style.overflow = overflow;
+      window.removeEventListener("keydown", handleKeyDown, false);
+    };
+  }, [hideModal]);
+
+  useLayoutEffect(
+    () => setPortal(document.querySelector(container)),
+    [container],
+  );
+
+  useEffect(() => {
+    if (fadeIn && ref.current) {
+      ref.current.focus();
+    }
+  }, [fadeIn]);
+
+  return (
+    portal &&
+    createPortal(
+      <div
+        className={twMerge(
+          "fixed top-0 right-0 flex h-screen w-110 items-center justify-center bg-transparent p-5 transition-opacity duration-150",
+          fadeIn ? "opacity-100" : "opacity-0",
+          "focus:outline-none",
+        )}
+        ref={ref}
+        tabIndex={-1}
+      >
+        <Box
+          className={twMerge(
+            "z-10 h-full w-full flex-col gap-4 shadow-green-600 transition-transform duration-150",
+            fadeIn ? "translate-x-0" : "translate-x-full",
+          )}
+          shadow="xl"
+        >
+          <div className="flex w-full justify-between border-green-600 border-b">
+            <Text.Heading>connect wallet</Text.Heading>
+            <Button.Tertiary icon="times" onClick={handleClose} />
+          </div>
+          <div className="flex w-full flex-col gap-4">
+            {installedWallets.map((wallet) => (
+              <WalletListItem
+                connecting={
+                  selectedWallet?.adapter.name === wallet.adapter.name &&
+                  connecting
+                }
+                handleClick={(event) =>
+                  handleWalletClick(event, wallet.adapter.name)
+                }
+                key={wallet.adapter.name}
```

Comment:
If wallets might have the same adapter name, React's `useId` can be very helpful! https://react.dev/reference/react/useId